src/table/use-sticky-state.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { useState, useCallback, useEffect } from 'react';
+import { useStickyColumnsContext } from './sticky-columns-context';
+
+export function useStickyState(isLastStickyLeft = false, isLastStickyRight = false) {
+  const [isStuckToTheLeft, setIsStuckToTheLeft] = useState(false);
+  const [isStuckToTheRight, setIsStuckToTheRight] = useState(false);
+
+  const leftCallback = useCallback(
+    ({ left }) => {
+      setIsStuckToTheLeft(left);
+    },
+    [setIsStuckToTheLeft]
+  );
+
+  const rightCallback = useCallback(
+    ({ right }) => {
+      setIsStuckToTheRight(right);
+    },
+    [setIsStuckToTheRight]
+  );
+
+  const { subscribe, unsubscribe } = useStickyColumnsContext();
+  useEffect(() => {
+    if (isLastStickyLeft) {
+      subscribe(leftCallback);
+    } else if (isLastStickyRight) {
+      subscribe(rightCallback);
+    }
+    return () => {
+      unsubscribe(leftCallback);
+      unsubscribe(rightCallback);
+    };
+  }, [isLastStickyLeft, isLastStickyRight, subscribe, leftCallback, rightCallback, unsubscribe]);
```

Comment:
The `subscribe` and `unsubscribe` functions are not memoized so the effect is triggered for every render. E.g. when the user resizes a column there are hundreds of subscribe calls. I am not sure if it impacts the performance much, but what can happen is that the in case of window resize when render and subscribe happen at the same time the callbacks update might be sent before all callbacks are set.

I think the subscribe/unsubscribe should only happen one when the cell mounts/un-mounts which can be secured with having an empty dependency array.

Reply:
Good call. I'll leave only isLastStickyLeft and isLastStickyRight here.